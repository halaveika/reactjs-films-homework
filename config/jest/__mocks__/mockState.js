const mockState = {
  content: {
    results: [{
      id: 2412412,
      title: 'Fight club',
      genre_ids: [13, 20, 21],
      vote_average: 8,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    },
    {
      id: 41412,
      title: 'Sopranos',
      genre_ids: [13, 18],
      vote_average: 10,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    }],
    page: 1,
    total_pages: 12,
    total_results: 55,
    genres_array: [{
      id: 13,
      name: 'Adventure',
    }, {
      id: 18,
      name: 'Drama',
    }, {
      id: 20,
      name: 'Family',
    }, {
      id: 21,
      name: 'Fantasy',
    }],
    video_url: 'test_url',
    details: {
      id: 2412412,
      title: 'Fight club',
      runtime: 120,
      vote_average: 10,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing',
    },
    isInitialisated: false,
  },
};

export default mockState
