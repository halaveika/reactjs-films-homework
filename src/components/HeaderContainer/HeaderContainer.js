import { connect } from 'react-redux';
import Header from '../Header';
import { setCurrentPage, setFilter, setSearchValue} from '../../modules/content/contentActions';
import { getPageSizeSelector } from '../../modules/content/contentSelectors';

export const mapStateToProps = (state) => ({
  pageSize: getPageSizeSelector(state)
});

const HeaderContainer = connect(mapStateToProps,
  { setCurrentPage, setFilter,setSearchValue })(Header);

export default HeaderContainer;

