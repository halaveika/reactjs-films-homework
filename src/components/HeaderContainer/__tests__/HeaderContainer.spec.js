import { mapStateToProps } from '../HeaderContainer';
import mockState from '../../../../config/jest/__mocks__/mockState';
describe('HeaderContainer', () => {
  it('should show mapStateToProps value', () => {
    expect(mapStateToProps(mockState)).toMatchSnapshot();
  });
});
