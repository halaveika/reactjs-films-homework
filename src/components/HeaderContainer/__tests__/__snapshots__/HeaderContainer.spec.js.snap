// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer should show mapStateToProps value 1`] = `
Object {
  "pageSize": 20,
}
`;
